### PR TITLE
fix: inner references not being found

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes the issue where inner references within models are not found.

**Related issue(s)**
fixes https://github.com/asyncapi/modelina/issues/841
blocked by https://github.com/asyncapi/modelina/pull/842

### CODE REVIEW

**Review by magicmatatjahu — DISMISSED** [review]
You probably will have merge conflicts due to this PR https://github.com/asyncapi/modelina/pull/842 and as I see you will handle duplicated deps in mentioned PR.

File: src/models/ConstrainedMetaModel.ts [hunk 1]
```diff
@@ -48,20 +48,23 @@ export class ConstrainedTupleModel extends ConstrainedMetaModel {
...
+        //Lets check the non-reference model for dependencies
+        dependencyModels = [...dependencyModels, ...tupleModel.value.getNearestDependencies()];
       }
-    ).map((tupleModel) => {
-      return tupleModel.value = tupleModel.value as ConstrainedReferenceModel;
+    }
+    
+    dependencyModels = dependencyModels.map((model) => {
+      return model as ConstrainedReferenceModel;
     });
```
Comment (magicmatatjahu) [inline src/models/ConstrainedMetaModel.ts:63]: What?! 😄

Comment (magicmatatjahu) [reply src/models/ConstrainedMetaModel.ts:63]: You should cast returned type to the `ConstrainedReferenceModel[]` if you have errors.

Comment (jonaslagoni) [reply src/models/ConstrainedMetaModel.ts]: Brainfart removed the unused code 😄

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**